client/src/pages/sports-grid/grid.js

```diff
@@ -1,45 +1,57 @@
 'use client'
 
 import Table from 'react-bootstrap/Table';
-import TeamSelector from './team-selector';
-import PlayerInput from './player-input';
+import Spinner from 'react-bootstrap/Spinner';
+
 import React from 'react';
 
+import PlayerInput from './player-input';
+import TeamLogo from './team-logo';
+
+import { useFetch } from '../../utils/hooks';
+
 export default function Grid() {
+    const data = useFetch(
+        'http://localhost:8000/sports-grid/basketball'
+    );
 
-    return (
-        <Table variant="dark">
-            <thead>
-                <tr>
-                    <td></td>
-                    <td>
-                        <TeamSelector></TeamSelector>
-                    </td>
-                    <td>
-                        <TeamSelector></TeamSelector>
-                    </td>
-                    <td>
-                        <TeamSelector></TeamSelector>
-                    </td>
-                </tr>
-                <tr>
-                    <td><TeamSelector></TeamSelector></td>
-                    <td><PlayerInput></PlayerInput></td>
-                    <td><PlayerInput></PlayerInput></td>
-                    <td><PlayerInput></PlayerInput></td>
-                </tr>
+    const tableHeaders = [];
+    const tableRows = [];
+
+    if (!data) {
+        return(<div>
+            <Spinner animation="border" role="status">
+                <span className="visually-hidden">Loading...</span>
+            </Spinner>
+        </div>)
+    }
+
+    if (!!data.cols) {
+        data.cols.forEach(team => tableHeaders.push(<TeamLogo key={team.Name} source={team.Logo}/>))
+    }
+
+    if (!!data.rows) {
+        for (var i = 0; i < data.rows.length; i++) {
+            const team = data.rows[i];
+            tableRows.push(
                 <tr>
-                    <td><TeamSelector></TeamSelector></td>
-                    <td><PlayerInput></PlayerInput></td>
-                    <td><PlayerInput></PlayerInput></td>
-                    <td><PlayerInput></PlayerInput></td>
+                    <TeamLogo key={team.Name} source={team.Logo}/>
+                    <td><PlayerInput pos={{col: 0, row: i}}></PlayerInput></td>
+                    <td><PlayerInput pos={{col: 1, row: i}}></PlayerInput></td>
+                    <td><PlayerInput pos={{col: 2, row: i}}></PlayerInput></td>
                 </tr>
+            );
+        }
+    }
+
+    return (
+        <Table variant="white" bordered={true}>
+            <thead>
                 <tr>
-                    <td><TeamSelector></TeamSelector></td>
-                    <td><PlayerInput></PlayerInput></td>
-                    <td><PlayerInput></PlayerInput></td>
-                    <td><PlayerInput></PlayerInput></td>
+                    <TeamLogo key={'NBA'} source={'https://cdn.ssref.net/req/202308101/tlogo/bbr/NBA.png'}/>
```

Comment:
dont think this will work forever. should save the image and serve it directly. also move the url to a constant.